tests/src/main/java/com/linkedin/kmf/tests/BasicPipelineTest.java

```diff
@@ -0,0 +1,293 @@
+import com.linkedin.kmf.services.configs.ProduceServiceConfig;
+import com.linkedin.kmf.services.configs.TopicManagementServiceConfig;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static net.sourceforge.argparse4j.impl.Arguments.store;
+
+/*
+ * The BasicPipelineTest test is intended to monitor the health and performance of kafka brokers across a mirror maker
+ * pipeline. It creates one producer and one consumer with the given configuration, produces messages with increasing
+ * integer in the message String, consumes messages, and keeps track of number of lost messages, duplicate messages, e2e
+ * latency, throughput, etc. As well, BasicPipeLineTest creates a topic management service for every kafka and zookeeper
+ * broker in the pipeline.
+ *
+ * BasicPipelineTest test exports these metrics via JMX. It also periodically report metrics if DEBUG level logging
+ * is enabled. This information can be used by other application to trigger alert when kafka brokers fail. It also
+ * allows users to track pipeline performance, e.g. latency and throughput.
+ */
+
+public class BasicPipelineTest implements Test {
+  private static final Logger LOG = LoggerFactory.getLogger(BasicPipelineTest.class);
+
+  private final ProduceService _produceService;
+  private final ConsumeService _consumeService;
+  private final List<TopicManagementService> _topicManagementServices;
+  private final String _name;
+
+  public BasicPipelineTest(Map<String, Object> props, String name) throws Exception {
+    _name = name;
+    _produceService = new ProduceService(createProduceServiceConfig(props), name);
+    _consumeService = new ConsumeService(createConsumeServiceConfig(props), name);
+    _topicManagementServices = new ArrayList<>();
+    createTopicManagementServices(props, name);
+  }
+
+  private void createTopicManagementServices(Map<String, Object> props, String name) {
+    Map<String, Object> topicManagementConfig = createTopicManagementServiceConfig(props);
+
+    List<String> zookeeperUrlsList = getList(topicManagementConfig, ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
+    if(zookeeperUrlsList == null) {
+      throw new RuntimeException(_name + ": Must specify at least one zookeeper connection.");
+    }
+
+    zookeeperUrlsList.add(getString(props, ProduceServiceConfig.ZOOKEEPER_CONNECT_CONFIG));
+    zookeeperUrlsList.add(getString(props, ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG));
+
+    Set<String> zookeeperUrls = new HashSet<>();
+    zookeeperUrls.addAll(zookeeperUrlsList);
```

Comment:
Instead of using a hashset to remove duplicates, it is probably better to just throw exception if user has provided a cluster more than once in this config.